### PR TITLE
docs: fixed some documentation errors in BLAKE2b

### PR DESCRIPTION
**Pull request by dsmurrow**
#### Description of Change

Removed doc comments describing include's as they seemed to mess up the doc generation. Also removed the define specifier in doc comments as Doxygen couldn't seem to parse them.

EDIT: I noticed a small bug in the algorithm which I fixed and added as a commit here
EDIT2: Added more tests for the algorithm

#### Checklist

- [x] Added description of change
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Review by alexpantyukhin — DISMISSED** [review]

File: hash/hash_blake2b.c [hunk 1]
```diff
@@ -14,10 +14,10 @@
  * 128 bits in length. Keyed hashing is also supported for
  * keys up to 64 bytes in length.
  */
-#include <assert.h>    /// for asserts
-#include <inttypes.h>  /// for fixed-width integer types e.g. uint64_t and uint8_t
-#include <stdio.h>     /// for IO
-#include <stdlib.h>    /// for malloc, calloc, and free. As well as size_t
+#include <assert.h>    // for asserts
```
Comment (Panquesito7) [inline hash/hash_blake2b.c:17]: Please add back the 3rd slash for all of the headers/libraries.

File: math/euclidean_algorithm_extended.c [hunk 2]
```diff
@@ -0,0 +1,148 @@
+/**
```
Comment (Panquesito7) [inline math/euclidean_algorithm_extended.c:1]: Please remove this file from the PR (it appears you already have a PR for this).

File: hash/hash_blake2b.c [hunk 3]
```diff
@@ -476,5 +467,75 @@ int main()
 
     free(digest);
 
+    uint8_t key[64] = {
```
Comment (Panquesito7) [inline hash/hash_blake2b.c:470]: Please add all of the tests in a separate test function to keep `main` clean.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: hash/hash_blake2b.c [hunk 4]
```diff
@@ -476,5 +468,84 @@ int main()
...
+        0x42, 0xf1, 0x6e, 0x9a, 0xb0, 0xea, 0xd3, 0x3b, 0x87, 0xc9, 0x19,
+        0x68, 0xa6, 0xe5, 0x09, 0xe1, 0x19, 0xff, 0x07, 0x78, 0x7b, 0x3e,
+        0xf4, 0x83, 0xe1, 0xdc, 0xdc, 0xcf, 0x6e, 0x30, 0x22};
+
+    digest = blake2b(filled, 64, filled_key, 64, 64);
+    assert_bytes(filled_answer, digest, 64);
+
+    free(digest);
+
+    printf("all tests have successfully passed!\n");
```
Comment (Panquesito7) [inline hash/hash_blake2b.c:539]:
```suggestion
    printf("All tests have successfully passed!\n");
```

**Review by Panquesito7 — APPROVED** [review]
Thanks! 🚀

**Review by alexpantyukhin — APPROVED** [review]

**Pull request merged**